traversal/selector/selector.go

```diff
@@ -106,6 +106,12 @@ type Selector interface {
 	// Only "Matcher" clauses actually implement this in a way that ever returns "true".
 	// See the Selector specs for discussion on "matched" vs "reached"/"visited" nodes.
 	Decide(node datamodel.Node) bool
```

Comment:
Are we using decide any more? Should we just delete it? No one uses compile selectors outside of go-ipld-prime I think

Reply:
i don't see any uses of `Decide` in the go-ipld-prime directly before this PR either. I think the thought is to transition to Match as the signature that we actually need for these walks.

My preference would be to mark this as deprecated rather than turn this into a backwards-incompatible change.